lib/puppet/type/kernel_parameter.rb

```diff
@@ -37,4 +39,25 @@
   autorequire(:file) do
     self[:target]
   end
+
+  # title_patterns method for mapping titles to namevars for supporting
+  # composite namevars.
+  # https://github.com/puppetlabs/puppetlabs-java_ks/blob/5bc34745a6f86e0c4af495e0ad3559c82d57873a/lib/puppet/type/java_ks.rb#L209
+  def self.title_patterns
+    [
+      [
+        %r{^([^:]+)$},
```

Comment:
On second though, in ruby `^` & `$` match the beginning/end of a line, not the string… So this would match something like `":garbage:\nname\nmore:garbage"`. Suggested change:

```suggestion
        %r{\A([^:]+)\z},
```